### ISSUE TITLE
Feature request: PPO early stopping, important for training stability

### ISSUE THREAD

**Issue by ShaojieJiang** [huggingface/trl#232]
Hi TRL contributors,

Thanks a lot for creating this nice library.
While using TRL, I realised that an important feature is still missing, that's the early stopping during a PPO step.
Without this feature, the optimisation is prone to KL/loss spikes such as those mentioned in #101 .
In my experiments, I found this issue is alleviated through reducing the default num PPO epochs.

This feature is already implemented in the [OpenAI official PPO](https://github.com/openai/spinningup/blob/master/spinup/algos/pytorch/ppo/ppo.py#L269) and [AllenAI RL4LMs](https://github.com/allenai/RL4LMs/blob/main/rl4lms/algorithms/ppo/ppo.py#L279).

Looking forward to your feedback.

Kind regards,
Shaojie

**Comment by edbeeching** [comment]
@lvwerra @lewtun  this might improve the issues we were discussing yesterday, I can add this feature if you like?

**Comment by lvwerra** [comment]
Sounds good! Yes, I agree this would be an easy and useful feature to add.

**Comment by ShaojieJiang** [comment]
Thanks @edbeeching @lvwerra for your quick responses. Looking forward to the improvement.
